### ISSUE TITLE
Can't locate Bio/Root/Version.pm in @INC (you may need to install the Bio::Root::Version module) 

### ISSUE THREAD

**Issue by ramnageena11** [tseemann/prokka#566]
powersequence@CH-CB-POWERSEQ:~$ prokka --help
Can't locate Bio/Root/Version.pm in @INC (you may need to install the Bio::Root::Version module) (@INC contains: /home/powersequence/anaconda3/bin/../perl5 /home/powersequence/anaconda3/lib/site_perl/5.26.2/x86_64-linux-thread-multi /home/powersequence/anaconda3/lib/site_perl/5.26.2 /home/powersequence/anaconda3/lib/5.26.2/x86_64-linux-thread-multi /home/powersequence/anaconda3/lib/5.26.2 .) at /home/powersequence/anaconda3/bin/prokka line 32.
BEGIN failed--compilation aborted at /home/powersequence/anaconda3/bin/prokka line 32.

I had run the command
sudo cpan Bio::Perl
and it was succesful.

Pls help me rectify the issue.

Thanks
Ram

**Comment by ramnageena11** [comment]
a new error is coming makeblastdb upgarde required

**Comment by benrhodes26** [comment]
I also have this issue.

**Comment by andersgs** [comment]
You have to make sure you have BioPerl version 1.72 or above installed. Can you try updating/upgrading BioPerl? It seems the earlier recipes on Conda have a bug.

**Comment by ramnageena11** [comment]
> I also had this issue, when using a conda (version 4.10.1) install into an existing environment. I found that if I created a new environment (i.e. conda create -n my_prokka_env) then I didn't experience this issue.
> 
> Alternatively, see this [comment](https://github.com/tseemann/prokka/issues/519#issuecomment-737076300) which provides a possible solution based on the perl version

Thanks for the suggestions, i will try this.

**Comment by ramnageena11** [comment]
> You have to make sure you have BioPerl version 1.72 or above installed. Can you try updating/upgrading BioPerl? It seems the earlier recipes on Conda have a bug.

Yes, I have updated the and upgraded Bioperl. upgarde did not executed but updated.

**Comment by andersgs** [comment]
@ramnageena11 is it ok to close the issue then?

**Comment by ramnageena11** [comment]
i checked the Bioperl version is 1.007002
It did not updated how could i do that?

**Comment by andersgs** [comment]
Did you try: 

`conda install -c bioconda perl-bioperl=1.72`

**Comment by andersgs** [comment]
Sorry. I had the incorrect version number. I edited the original post to fix that.

**Comment by ramnageena11** [comment]
Hey andersgs,
I think that worked but now
[13:49:40] This is prokka 1.14.6
[13:49:40] Written by Torsten Seemann <torsten.seemann@gmail.com>
[13:49:40] Homepage is https://github.com/tseemann/prokka
[13:49:40] Local time is Fri Apr 30 13:49:40 2021
[13:49:40] You are powersequence
[13:49:40] Operating system is linux
[13:49:40] You have BioPerl 1.007002
[13:49:40] System has 28 cores.
[13:49:40] Will use maximum of 8 cores.
[13:49:40] Annotating as >>> Bacteria <<<
[13:49:40] The sequence databases have not been indexed. Please run 'prokka --setupdb' first.

**Comment by andersgs** [comment]
Follow the instructions on the last line and you should be good to go

**Comment by ramnageena11** [comment]
(my_prokka_env) powersequence@CH-CB-POWERSEQ:~/prokka/bin$ prokka --setupdb

Command 'prokka' not found, did you mean:

  command 'propka' from deb pdb2pqr

**Comment by RxLoutre** [comment]
Had the same issues : 

` streptidy git:(master) ✗ prokka -h
Can't locate Bio/Root/Version.pm in @INC (you may need to install the Bio::Root::Version module) (@INC contains:`

And I directly tried what @andersgs suggested : `conda install -c bioconda perl-bioperl=1.7.2`

And it solves the issue for me. At least I can run prokka -h :)

**Comment by ramnageena11** [comment]
hi andersgs
system got crashed. I am re-installing.
will update.
Thanks

**Comment by ramnageena11** [comment]
Hi,
Pls see and suggest.

(base) powersequencing@CH-CB-RS-P5820:~$ prokka --version
Can't locate Bio/Root/Version.pm in @INC (you may need to install the Bio::Root::Version module) (@INC contains: /home/powersequencing/anaconda3/bin/../perl5 /home/powersequencing/anaconda3/lib/site_perl/5.26.2/x86_64-linux-thread-multi /home/powersequencing/anaconda3/lib/site_perl/5.26.2 /home/powersequencing/anaconda3/lib/5.26.2/x86_64-linux-thread-multi /home/powersequencing/anaconda3/lib/5.26.2 .) at /home/powersequencing/anaconda3/bin/prokka line 32.
BEGIN failed--compilation aborted at /home/powersequencing/anaconda3/bin/prokka line 32.
(base) powersequencing@CH-CB-RS-P5820:~$ ^C
(base) powersequencing@CH-CB-RS-P5820:~$ conda install -c bioconda perl-bioperl=1.7.2
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: - 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                                                                                                                                                                             

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package zlib conflicts for:
python=3.9 -> zlib[version='>=1.2.11,<1.3.0a0']
perl-bioperl=1.7.2 -> perl-bio-samtools -> zlib[version='>=1.2.11,<1.3.0a0']The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.27=0
  - python=3.9 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']

Your installed version is: 2.27

**Comment by ramnageena11** [comment]
Hi andersgs,
could you suggest, how to resolve?

Thanks
ram

**Comment by ramnageena11** [comment]
Hi,
Its done.

Thanks
